cmd/queuebench/main.go

```diff
@@ -200,6 +200,11 @@ func produce(ctx context.Context, p *kafka.Producer, topic apmqueue.Topic, size
 
 	deadline := time.Now().Add(duration)
 	for time.Now().Before(deadline) {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
```

Comment:
Is this necessary? Does the Produce method below not honour the context cancellation?

Reply:
The `Produce` method does not but context cancellation is handled by `franz-go/kgo` (see [docs](https://github.com/twmb/franz-go/blob/89cd6a7c4fef29a400bba5a6cd8a77def5ec8105/pkg/kgo/producer.go#L326-L356)), I removed it.